scipy/interpolate/tests/test_interpolate.py

```diff
@@ -1341,6 +1341,12 @@ def test_yi_trailing_dims(self):
         pp = BPoly.from_derivatives(xi, yi)
         assert_equal(pp.c.shape, (2*k, m, 6, 7, 8))
 
+    def test_gh_5430(self):
+        orders = np.int64(1)
```

Comment:
Hm... I'll test further. I don't think it's non-native versus native; for me int64 fails in py3k and int32 fails in py2k. (FWIW my system is 64 bit.)